samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -690,3 +694,46 @@ int link_rai_enable(bool enable)
 	}
 	return 0;
 }
+
+int link_setdnsaddr(const char *ip_address)
+{
+	struct nrf_in_addr in4_addr;
+	struct nrf_in6_addr in6_addr;
+	int family = NRF_AF_INET;
+	void *in_addr = NULL;
+	nrf_socklen_t in_size = 0;
+	int ret = 0;
+
+	if (strlen(ip_address) > 0) {
+		in_addr = &in4_addr;
+		in_size = sizeof(in4_addr);
+		ret = nrf_inet_pton(family, ip_address, in_addr);
```

Comment:
why not inet_pton()?

Reply:
When testing I found that `inet_pton()` would happily accept strings like "1" as a valid IP address, while `nrf_inet_pton()` is more correct. Unfortunate we don't have any socket offloading for this functions. Have a look in `net_utils_ip_string_is_valid()` where I added a comment for this.

Reply:
ok